src/encord_active/app/project_comparison/project_similarity.py

```diff
@@ -0,0 +1,165 @@
+        st.write("There is no compatible metric")
+        return
+
+    metric_name_2 = metric_selection_col_2.selectbox(
+        "Select the second metric",
+        options=options_for_second_metric,
+        key=f"project_comparison_metric_selection_2_{get_state().project_paths.project_dir.name}_{project_name_2}",
+    )
+
+    if metric_name_1 == metric_name_2:
+        st.warning("Please choose a different metric")
+        return
+
+    project_values_1 = merged_metrics_1[[metric_name_1, metric_name_2]].copy().dropna()
+    project_name_1 = get_state().project_paths.project_dir.name
+    project_values_1["project"] = project_name_1
+
+    # Check if the selected metrics are also available in the second project
+    if metric_name_1 not in merged_metrics_2 or merged_metrics_2[metric_name_1].dropna().shape[0] == 0:
+        st.write(f"Metric **{metric_name_1}** is not available for the project `{project_name_2}`.")
+        return
+
+    if metric_name_2 not in merged_metrics_2 or merged_metrics_2[metric_name_2].dropna().shape[0] == 0:
+        st.write(f"Metric **{metric_name_2}** is not available for the project `{project_name_2}`.")
+        return
+
+    project_values_2 = merged_metrics_2[[metric_name_1, metric_name_2]].copy().dropna()
+    project_values_2["project"] = project_name_2
+
+    # TODO change append to concat
+    project_values = pd.concat([project_values_1, project_values_2], ignore_index=True)
+
+    fig = render_2d_metric_similarity_plot(
+        project_values, cast(str, metric_name_1), cast(str, metric_name_2), project_name_1, project_name_2
+    )
+    st.plotly_chart(fig, use_container_width=True)
+
+
+def project_similarity():
+    def render():
+        setup_page()
+
+        project_metas = {project: fetch_project_meta(project) for project in project_list(get_state().target_path)}
+        project_metas = {k: v for k, v in sorted(project_metas.items(), key=lambda item: item[0])}
+
+        selected_project = st.selectbox(
+            "Select project to compare",
+            options=project_metas,
+            index=0,
+            format_func=(lambda x: project_metas[x]["project_title"]),
+            key=f"project_similarity_{get_state().project_paths.project_dir.name}",
+        )
+
+        # TODO this is a hacky way to get MergedMetrics of another project, it should be fixed later
+        DBConnection.set_project_file_structure(ProjectFileStructure(selected_project))
+        merged_metrics_2 = MergedMetrics().all()
+        project_name_2 = project_metas[selected_project]["project_title"]
+        DBConnection.set_project_file_structure(ProjectFileStructure(get_state().project_paths.project_dir))
+
+        all_metrics = sorted([metric for metric in fetch_metrics_meta(get_state().project_paths)])
+
+        metric_similarities = {"metric": [], "similarity_score": []}
+        for metric in all_metrics:
+            if metric in METRICS_TO_EXCLUDE:
+                continue
```

Comment:
Is it on purpose that we don't remove the metric from `all_metrics`?
The consequence is that the 2D plot will still allow you to choose, e.g., random as one of the axes.

Reply:
Yeah, it is on purpose. These excluded metrics would affect the average similarity score in not a meaningful way so I excluded them but thought the user might still want to see their distributions.